src/test/java/org/minions/devfund/damianvp/battleship/OceanTest.java

```diff
@@ -0,0 +1,254 @@
+package org.minions.devfund.damianvp.battleship;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * OceanTest class.
+ */
+public class OceanTest {
+    private Ocean ocean;
+    static final int LENGTH = 20;
+    /**
+     * Method to initialize each test.
+     */
+    @Before
+    public void setOceanClassTest() {
+        ocean = new Ocean();
+    }
+
+    /**
+     * Method to test the Ocean constructor.
+     */
+    @Test
+    public void testConstructor() {
+        assertEquals(0, ocean.getHitCount());
+        assertEquals(0, ocean.getShipsSunk());
+        assertEquals(0, ocean.getShotsFired());
+        for (int i = 0; i < LENGTH; i++) {
+            for (int j = 0; j < LENGTH; j++) {
+                assertEquals("empty", ocean.getShipArray()[i][j].getShipType());
+            }
+        }
+    }
+
+    /**
+     * Method to test placeAllShipsRandomly.
+     */
+    @Test
+    public void testPlaceAllShipsRandomly() {
+//        Ocean ocean = new Ocean();
+        ocean.placeAllShipsRandomly();
+        // 8 + 7 + 2*6 + 2*5 + 3*4 + 4*3 = 61
+        final int expectedCount = 61;
+        assertEquals(0, ocean.getHitCount());
+        assertEquals(0, ocean.getShipsSunk());
+        assertEquals(0, ocean.getShotsFired());
+        int shipCount = 0;
+        for (int i = 0; i < LENGTH; i++) {
+            for (int j = 0; j < LENGTH; j++) {
+                if (!ocean.getShipArray()[i][j].getShipType().equals("empty")) {
+                    shipCount++;
+                }
+            }
+        }
+        assertEquals(expectedCount, shipCount);
+    }
+
+    /**
+     * Method to test isOccupied method in empty ocean.
+     */
+    @Test
+    public void testIsOccupiedInEmptyOcean() {
+        for (int i = 0; i < LENGTH; i++) {
+            for (int j = 0; j < LENGTH; j++) {
+                assertFalse(ocean.isOccupied(i, j));
+                i++;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)